sw/device/silicon_creator/manuf/data/manuf_testplan.hjson

```diff
@@ -0,0 +1,408 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+{
+  name: "manufacturing"
+
+  testpoints: [
+    {
+      name: manuf_cp_unlock_raw
+      desc: '''Verify transition from RAW to TEST_UNLOCKED lc_state.
+
+            - Pre-load OTP with RAW lc_state.
+            - Bring device out of reset selecting the Life Cycle (LC) TAP interface using strap pins.
+
+            Perfom the following steps via LC TAP interface:
+
+            - Switch to external clock via TRANSITION_CTRL.EXT_CLOCK_EN.
+            - Provide RAW_UNLOCK.
+            - Poll for lc_ctrl STATUS register for TRANSITION_SUCCESSFUL. Check for errors or timeout.
+            - Perform reset by toggling the POR pin.
+            - Verify the lc_state is TEST_UNLOCKED0.
+
+            The manufacturing environment is required to support test reruns whenever possible:
+
+            - Verify that the RAW_UNLOCK process can be re-tried after failed or interrupted attempts.
+            '''
+      tags: ["dv", "fpga", "silicon"]
+      stage: V3
+      tests: []
+    }
+
+    {
+      name: manuf_scrap
+      desc: '''Verify transition from any state to SCRAP lc_state.
+
+            Manufacturing tests should be able to move the device from any lc_state into SCRAP state.
+
+            For each manufacturing lc_state `s`:
+            - Pre-load OTP with `s` lc_state image.
+            - Bring device out of reset selecting the Life Cycle (LC) TAP interface using strap pins.
+            - Request transition to SCRAP state via LC TAP interface.
+            - Perform reset by togglig the POR pin.
+            - Check we are in SCRAP state via LC TAP interface.
+
+            The manufacturing environment is required to support test reruns whenever possible:
+
+            - Verify that the transition to SCRAP mode can be re-tried after failed or interrupted attempts.
+            '''
+      tags: ["dv", "fpga", "silicon"]
+      stage: V3
+      tests: []
+    }
+
+    {
+      name: manuf_cp_yield_test
+      desc: '''Verify that it is possible to switch TAP interfaces while injecting external clock.
+
+            Verify that it is possible to access TAP interfaces while injecting an external clock.
+            This test case is valid only in test life cycle states.
+
+            - Pre-load OTP with TEST_UNLOCKED lc_state in pre-silicon environments. Advance to
+              state in silicon.
+            - Enable external clock injection.
+            - Connect to CPU JTAG and perform register read/writes to memory/registers.
+            - Switch to LC TAP by manipulating the strap pins.
+            - Verify that read/writes to lc_ctrl work as expected.
+            '''
```

Comment:
@dmcardle told me that halting and debugging the core via JTAG is practically impossible when ROM is executing and resetting the chip because the next stage is invalid. Thus, any JTAG tests should probably assume they require `rom_exec_en = 0`.

Reply:
wait, i think we might need to discuss this specific point more.
Is there no way to halt the ibex core? It doesn't have to be exact, but we should have a way to say stop.

Reply:
> @dmcardle told me that halting and debugging the core via JTAG is practically impossible when ROM is executing and resetting the chip because the next stage is invalid. Thus, any JTAG tests should probably assume they require `rom_exec_en = 0`.

I'll walk this back a little bit. In #15024, I'm now able to halt the device despite execution being enabled. Admittedly, it is a little flaky. It fails in maybe 1/10 tries. If we allow for repeated attempts I suppose we could achieve an arbitrary flakiness tolerance.

Reply:
I think the issue is that the watchdog timer is configured. We tell it to halt and reset, and it obeys. But then the watchdog timer fires after everything was supposed to be reset, and OpenOCD errors because the device reset unexpectedly. I've added some GDB commands to reset the timer, but still racy because the timer might go off (a) while OpenOCD is establishing a connection or (b) while we're resetting the timer.

Reply:
hmm... i thought @alphan had set the rom watchdog to something really long, am I misremembering? 
If it helps, maybe we should consider extending a bit from `rv_dm` to the `aon_timer` to halt it automatically (in future release).  Otherwise having to deal with the watchdog every time is going to be frustrating if ever we need to rely on it.

Reply:
Oh no, I think you're right. I may have deluded myself while indirectly messing with timing. I'll have to look closer.

Reply:
this might warrant a separate issue..we may need to bring in the lowrisc folks to help have a look.

Reply:
@tjaychen yes, watchdog timer is set to 1 s [here](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/rom/rom_start.S;l=169?q=rom_start.&ss=opentitan%2Fopentitan). Since this value is hard-coded based on the clock frequencies we expect, I wonder if there could be a miscalculation somewhere if you guys still think watchdog timer is the culprit here.

Based on @dmcardle's comment above, it looks like we can attach a debugger, even though it's flaky. Please note that my original comment is almost a week old and @dmcardle has been actively working on the jtag/openOCD/gdb.